From: Nathan Chancellor <nathan@kernel.org>
Subject: hexagon: process: add internal prototype for do_work_pending()
Date: Thu, 30 Nov 2023 15:58:27 -0700

Clang warns:

  arch/hexagon/kernel/process.c:155:5: warning: no previous prototype for function 'do_work_pending' [-Wmissing-prototypes]
    155 | int do_work_pending(struct pt_regs *regs, u32 thread_info_flags)
        |     ^
  arch/hexagon/kernel/process.c:155:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
    155 | int do_work_pending(struct pt_regs *regs, u32 thread_info_flags)
        | ^
        | static

This function is only referenced from assembly, so it does not technically
need a prototype.  Add one right above the definition anyways to clear up
the warning.

Link: https://lkml.kernel.org/r/20231130-hexagon-missing-prototypes-v1-14-5c34714afe9e@kernel.org
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Cc: Arnd Bergmann <arnd@arndb.de>
Cc: Brian Cain <bcain@quicinc.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/hexagon/kernel/process.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/hexagon/kernel/process.c~hexagon-process-add-internal-prototype-for-do_work_pending
+++ a/arch/hexagon/kernel/process.c
@@ -153,6 +153,7 @@ unsigned long __get_wchan(struct task_st
  * Returns 0 if there's no need to re-check for more work.
  */
 
+int do_work_pending(struct pt_regs *regs, u32 thread_info_flags);
 int do_work_pending(struct pt_regs *regs, u32 thread_info_flags)
 {
 	if (!(thread_info_flags & _TIF_WORK_MASK)) {
_
